api/src/types/ast.ts

```diff
@@ -55,9 +55,20 @@ export type FunctionFieldNode = {
 	relatedCollection: string;
 };
 
+export type JsonFieldNode = {
+	type: 'jsonField';
+	name: string; // field string
+	fieldKey: string; // alias
```

Comment:
> Also would be nice to have a short description for temporary.

I had a comment for this at some point but having `"whether this is a temporary node"` on a property called `temporary` felt a bit redundant. If this is less obvious than i had assumed i could add a description like `"temporary fields nodes do not end up in the final result"` 🤔

Reply:
> > Also would be nice to have a short description for temporary.
> 
> I had a comment for this at some point but having `"whether this is a temporary node"` on a property called `temporary` felt a bit redundant. If this is less obvious than i had assumed i could add a description like `"temporary fields nodes do not end up in the final result"` 🤔

For myself it was not quite clear for _what_ temporary nodes are needed in the first place, so I was hoping that maybe this could be made a little clearer with a comment. 😄